src/server.rs

```diff
@@ -322,7 +322,7 @@ impl HttpServer {
     /// full capacity has already been reached.
     /// `InvalidWrite` is returned when the server attempted to perform a write operation
     /// on a connection on which it is not possible.
-    pub fn requests(&mut self) -> Result<Vec<ServerRequest>> {
+    pub fn requests(&mut self, maybe_max_len: Option<usize>) -> Result<Vec<ServerRequest>> {
```

Comment:
Let's specify the limit when we create the server instead of doing it every single time we try to receive requests. TI think it fits much better in `HttpServer::new`.